Add Module dependencies attribute (#1259)
PrestaShop/docs: modules/concepts/module-class/_index.md

This attribute contains a confirmation message to be displayed when a user attempts to uninstall this module.

### dependencies

```php
$this->dependencies = ['ps_googleanalytics', 'productcomments'];
```

- **Type:** array
- **Default:** (empty)
- **Required:** No

> This attribute contains the module's dependencies that are checked at install. if one of the dependant modules are not installed, the installation will fails and a warning displayed.

Reviewer: ```suggestion
This attribute contains the module's dependencies that PrestaShop checks when installing the module. If one of the dependant modules is unavailable, the installation will fail, and PrestaShop will display a warning.
```
